code/game/objects/items/weaponry.dm

```diff
@@ -43,30 +43,41 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 
 /obj/item/balloon_mallet
 	name = "balloon mallet"
-	desc = "It's a mallet, a weapon known for being heavy, but made from notoriously light balloons. Air inside removes any force from the swings. It'd be quite embarrassing to get hit by this."
+	desc = "A sacred weapon of the higher castes from the clown planet, used to strike fear into the hearts of their foes. Wield it with care. It'd be quite embarrassing to get hit by this."
 	icon = 'icons/obj/weapons/hammer.dmi'
-	icon_state = "balloon_mallet"
-	inhand_icon_state = "balloon_mallet"
+	icon_state = "balloon_mallet0"
+	base_icon_state = "balloon_mallet"
+	worn_icon_state = "balloon_mallet"
 	lefthand_file = 'icons/mob/inhands/weapons/hammers_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/weapons/hammers_righthand.dmi'
+	slot_flags = ITEM_SLOT_BACK
 	siemens_coefficient = 0
 	force = 1
 	throw_speed = 1
 	throwforce = 1
-	throw_range = 1
+	throw_range = 7
 	w_class = WEIGHT_CLASS_HUGE
 	attack_verb_continuous = list("mallets", "smoother")
 	attack_verb_simple = list("mallet", "smoother")
 	max_integrity = 20
 	armor_type = /datum/armor/item_banhammer
 	resistance_flags = FIRE_PROOF
 
-/obj/item/balloon_mallet/examine(mob/user)
+/obj/item/balloon_mallet/Initialize(mapload)
 	. = ..()
-	if(HAS_TRAIT(user,TRAIT_BALLOON_SUTRA))
-		. = "A sacred weapon of the higher castes from the clown planet, used to strike fear into the hearts of their foes. Wield it with care."
+	AddElement(/datum/element/knockback, 10, FALSE, TRUE)
+	AddComponent(/datum/component/boomerang, throw_range+7, TRUE)
+	AddComponent(/datum/component/two_handed, \
+		icon_wielded = "[base_icon_state]1", \
+		attacksound = SFX_CLOWN_STEP, \
+	)
+
+/obj/item/balloon_mallet/update_icon_state()
+	icon_state = "[base_icon_state]0"
+	return ..()
 
 /obj/item/balloon_mallet/attack(mob/living/target, mob/living/user)
+	..()
```

Comment:
This return that you have added will break this proc (because it already calls parent later, if user is in combat mode)